packages/legend-query-builder/src/stores/QueryLoaderState.ts

```diff
@@ -0,0 +1,281 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DEFAULT_TYPEAHEAD_SEARCH_MINIMUM_SEARCH_LENGTH,
+  DEFAULT_TYPEAHEAD_SEARCH_LIMIT,
+  type GenericLegendApplicationStore,
+} from '@finos/legend-application';
+import {
+  type LightQuery,
+  QuerySearchSpecification,
+  type AbstractPureGraphManager,
+  type QueryProjectCoordinates,
+  V1_PureGraphManager,
+  type Query,
+  type QueryInfo,
+} from '@finos/legend-graph';
+import {
+  ActionState,
+  type GeneratorFn,
+  assertErrorThrown,
+  guaranteeType,
+  deleteEntry,
+  guaranteeNonNullable,
+} from '@finos/legend-shared';
+import { makeObservable, observable, action, flow, flowResult } from 'mobx';
+import type { QueryBuilderState } from './QueryBuilderState.js';
+import type {
+  LoadQueryFilterOption,
+  QueryBuilder_LegendApplicationPlugin_Extension,
+} from './QueryBuilder_LegendApplicationPlugin_Extension.js';
+
+export class QueryLoaderState {
+  readonly applicationStore: GenericLegendApplicationStore;
+  loadQueriesState = ActionState.create();
+  queries: LightQuery[] = [];
+  isQueryLoaderOpen = false;
+  searchText = '';
+  showCurrentUserQueriesOnly = false;
+  showPreviewViewer = false;
+  queryPreviewContent?: QueryInfo;
+  extraFilters = new Map<string, boolean>();
+  extraFilterOptions: LoadQueryFilterOption[] = [];
+  searchSpecificationProjectCoordinates?: QueryProjectCoordinates[];
+  currentUserId: string | undefined;
+  queryBuilderState?: QueryBuilderState | undefined;
+  persistedQueries: string[];
+  removeQueryId?:
+    | ((
+        persistedQueries: string[],
+        idx: number,
+        applicationStore: GenericLegendApplicationStore,
+      ) => void)
+    | undefined;
+  persistQueryIds?:
+    | ((
+        persistedQueries: string[],
+        val: LightQuery | Query,
+        applicationStore: GenericLegendApplicationStore,
+      ) => void)
+    | undefined;
+
+  constructor(
+    applicationStore: GenericLegendApplicationStore,
+    persistedQueries?: string[],
```

Comment:
let's make this a little more generic, with the way they are named/used right now, it makes it awfully clear that we have these set for `Legend Query`, perhaps, we can shape it so that we have `getInitialQueries`, `onLoadQuery`, `onDeleteQuery` instead?